helm/vitess/templates/_orchestrator-conf.tpl

```diff
@@ -0,0 +1,163 @@
+    "MySQLTopologyCredentialsConfigFile": "",
+    "MySQLTopologyMaxPoolConnections": 3,
+    "MySQLTopologyPassword": "orc_client_user_password",
+    "MySQLTopologyReadTimeoutSeconds": 3,
+    "MySQLTopologySSLCAFile": "",
+    "MySQLTopologySSLCertFile": "",
+    "MySQLTopologySSLPrivateKeyFile": "",
+    "MySQLTopologySSLSkipVerify": true,
+    "MySQLTopologyUseMutualTLS": false,
+    "MySQLTopologyUser": "orc_client_user",
+    "OnFailureDetectionProcesses": [
+        "echo 'Detected {failureType} on {failureCluster}. Affected replicas: {countSlaves}' >> /tmp/recovery.log"
+    ],
+    "OSCIgnoreHostnameFilters": [
+    ],
+    "PhysicalEnvironmentPattern": "[.]([^.]+[.][^.]+)[.]mydomain[.]com",
+    "PostFailoverProcesses": [
+        "echo '(for all types) Recovered from {failureType} on {failureCluster}. Failed: {failedHost}:{failedPort}; Successor: {successorHost}:{successorPort}' >> /tmp/recovery.log"
+    ],
+    "PostIntermediateMasterFailoverProcesses": [
+        "echo 'Recovered from {failureType} on {failureCluster}. Failed: {failedHost}:{failedPort}; Successor: {successorHost}:{successorPort}' >> /tmp/recovery.log"
+    ],
+    "PostMasterFailoverProcesses": [
+        "echo 'Recovered from {failureType} on {failureCluster}. Failed: {failedHost}:{failedPort}; Promoted: {successorHost}:{successorPort}' >> /tmp/recovery.log",
+        "vtctlclient -server vtctld.{{ $namespace }}:15999 TabletExternallyReparented {successorAlias}"
+    ],
+    "PostponeSlaveRecoveryOnLagMinutes": 0,
+    "PostUnsuccessfulFailoverProcesses": [
+    ],
+    "PowerAuthUsers": [
+        "*"
+    ],
+    "PreFailoverProcesses": [
+        "echo 'Will recover from {failureType} on {failureCluster}' >> /tmp/recovery.log"
+    ],
+    "ProblemIgnoreHostnameFilters": [
+    ],
+    "PromotionIgnoreHostnameFilters": [
+    ],
+    "PseudoGTIDMonotonicHint": "asc:",
+    "PseudoGTIDPattern": "drop view if exists .*?`_pseudo_gtid_hint__",
+    "RaftAdvertise": "POD_NAME.{{ $namespace }}",
+    "RaftBind": "POD_NAME",
+    "RaftDataDir": "/var/lib/orchestrator",
+    "RaftEnabled": true,
+    "RaftNodes": [
+    {{ range $i := until (int $orc.replicas) }}
+        "orchestrator-{{ $i }}.{{ $namespace }}"{{ if lt $i (sub (int64 $orc.replicas) 1) }},{{ end }}
+    {{ end }}
+    ],
+    "ReadLongRunningQueries": false,
+    "ReadOnly": false,
+    "ReasonableMaintenanceReplicationLagSeconds": 20,
+    "ReasonableReplicationLagSeconds": 10,
+    "RecoverIntermediateMasterClusterFilters": [
+        "_intermediate_master_pattern_"
+    ],
+    "RecoverMasterClusterFilters": [
+        ".*"
+    ],
+    "RecoveryIgnoreHostnameFilters": [
+    ],
+    "RecoveryPeriodBlockMinutes": 1,
+    "RecoveryPeriodBlockSeconds": 60,
+    "ReduceReplicationAnalysisCount": true,
+    "RejectHostnameResolvePattern": "",
+    "RemoveTextFromHostnameDisplay": ".mydomain.com:3306",
+    "ReplicationLagQuery": "select round(absolute_lag) from meta.heartbeat_view",
```

Comment:
Is there a heartbeat injection in Vitess?

Reply:
No, Vitess doesn't have anything built-in AFAIK.

Reply:
Vitess does have a heartbeat built in, but it is not enabled by default. We have it enabled on all of our tablets however.

Reply:
@bbeaudreault Would you mind sending me what you use for the heartbeat query?

Reply:
Along with whatever flag is used to enable it?

Reply:
If it's enabled, I believe it should look something like this:

```sql
SELECT unix_timestamp() - floor(ts/1000000000) FROM `_vt`.heartbeat WHERE keyspaceShard='KEYSPACE:SHARD';
```

It can be enabled with `-heartbeat_enable`, and the interval can be configured with `-heartbeat_interval 1s`. The default is 1s.